consensus/common/src/main/java/org/hyperledger/besu/consensus/common/ForksSchedule.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Hyperledger Besu contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.common;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.NavigableSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Function;
+
+import com.google.common.annotations.VisibleForTesting;
+
+public class ForksSchedule<C> {
+
+  private final NavigableSet<ForkSpec<C>> forks =
+      new TreeSet<>(
+          Comparator.comparing((Function<ForkSpec<C>, Long>) ForkSpec::getBlock).reversed());
+
+  @VisibleForTesting
```

Comment:
Should remove this, it's not just for testing now

Reply:
Good spot 👍